### PR TITLE
[wip] implement email notifications in scholar 3

### LINKED ISSUE #886: Implement email notifications in Scholar 3
Need to review Scholar 2 email functionality and current Sufia notification/email decide what's important to port.

Includes:
- New user emails
- Change manager notification emails

### PR DESCRIPTION
**Pull request by lawhorkl**
Fixes #886

Implement change manager notifications in scholar 3 for:

* Adding/removing editors in works
* Adding/removing proxy users
* Embargos

### CODE REVIEW

**Comment by hortongn** [comment]
@lawhorkl My suggestions for changes:

Uncomment the methods in app/controllers/depositors_controller.rb so that emails are sent when a proxy is removed.

Add `.deliver` to the notify methods in lib/tasks/embargo_manager.rake

For the embargo reminder email body I'm seeing `This email is to inform you that your Scholar@UC item @work_title has reached its embargo date and is now available to the public.` You're missing `<%= %>` around `@work_title` in the notify view.

For the email users receive when a proxy is added/removed, change the text to 
`A proxy can create and upload files on behalf of another user as well as edit, or delete those corresponding works and files.  If you feel that this was made in error, please contact the Scholar@UC team.`

The above matches better with the description that's given for proxies on the web page.

**Comment by lawhorkl** [comment]
@hortongn ill update the branch asap

**Comment by lawhorkl** [comment]
@hortongn i uncommented the code and added a spec for the depositors controller from sufia to hopefully fix what happened before. I have to leave soon but will pick up on monday

**Comment by lawhorkl** [comment]
this is back to WIP for now

**Comment by hortongn** [comment]
@lawhorkl Updated list of changes needed:

You can fix your failing depositors spec test using:
`expect { delete :destroy, user_id: user.user_key, id: grantee.user_key, use_route: :sufia }.to change { ProxyDepositRights.count }.by(-1)`
(I had to add `use_route: : sufia` to get it to see the route from Sufia)

You still need to add `.deliver` to your notify methods in lib/tasks/embargo_manager.rake

When I get the embargo email, it still isn't showing me the title of work.  It says `This email is to inform you that your Scholar@UC item t has reached its embargo date and is now available to the public`

GitHub says you now have a conflict in .rubocopy.yml.  Rebase on develop and resolve the conflict.

File: app/helpers/change_manager/change_manager_helper.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+# helper for the WorksControllerBehavior class
+# cleaner to keep all change manager code in here
+module ChangeManager
+  module ChangeManagerHelper
+    def queue_notifications_for_editors(editors)
+      editors.each do |editor_wrapper|
+        editor = editor_wrapper[1]
```
Comment (lawhorkl) [inline app/helpers/change_manager/change_manager_helper.rb]: here

**Pull request merged**